src/routes/trip-members.js

```diff
@@ -89,11 +103,46 @@ export function leave (req, res) {
   if (!tripId) return res.sendStatus(400)
   if (req.user === owner) return res.sendStatus(400)
 
+  // check if user was approved
+  const { pending } = req.db.get('SELECT pending FROM trip_members WHERE trip = ? and user = ?', tripId, req.user)
+
   const { changes } = req.db.run('DELETE FROM trip_members WHERE trip = ? and user = ?',
     tripId, req.user)
   if (changes === 0) console.warn(`Unnecessary delete requested for trip ${tripId}`)
 
-  // TODO only send if user was approved
-  mailer.send(emails.getUserLeftEmail, req.db, tripId, req.user)
+  if (!pending) {
+    mailer.send(emails.getUserLeftEmail, req.db, tripId, req.user)
+    autoApprove(tripId)(req.db)
+  }
   return tripCard.renderSignupCard(req, res, tripId, req.user)
 }
+
+export function autoApprove (tripId) {
+  return (db) => {
+    let moreToApprove = true
+    while (moreToApprove) {
```

Comment:
I had a substantial comment that got lost somehow, sorry to add this after.

A while loop is not the way to go here, for two reasons:
1. If there's any possibility of the while loop not terminating, it will crash the program by blocking the event loop. The other two while loops in this code have much simpler terminating conditions, and writing this makes me think I should re-evaluate them, honestly.
2. Generally speaking, it is much simpler (and more efficient) to let SQL handle operations over multiple entries, in this case meaning "all the people who are pending admittance". Think functionally (operations over arrays) not procedurally (loops).

A better way to do it is the following:
1. If the time is less than 24 hours from now, return
2. Get the total number of members currently in the trip
3. Get the number of auto approved members
4. Let X be `auto_approved_members - total_members`. If X >= 0, get the pending trip members, limit X, ordered by `ROWID`
5. Admit them to the trip
6. Send them emails

No loops required (though a `map` or `forEach` is the simplest way to do the emails). I would do (4) and (5) as virtually identical statements, with (5) as an update; this is much easier than trying to do an array input to SQL. (2) and (3) can be combined into one statement pretty easily, if you want to test your SQL.